### ISSUE TITLE
Comparing Null Objects

### ISSUE THREAD

**Issue by Mansari** [GregFinzer/Compare-Net-Objects#72]
Hi. I am trying to use the library to log details of objects when added or removed. I am doing this by passing a null object along with the old or new object. I am getting the following userfriendly report though "Expected != Actual: (null) <<ChangedTo>> ...". Is there a way to change this text so I only show the old values or the new values?

**Comment by GregFinzer** [comment]
@Mansari

Thanks so much.

I would new up a blank object to compare and then set the Config.MaxDifferences to Int32.Max

**Comment by Mansari** [comment]
Thanks. One more thing, in comparing lists, ideally I would like to get a dump of what objects changed (i.e. old list had 3 members, new list has 4, which ones were removed, added, changed). The current input (for removed members looks like this - listing the item TYPE, which is not what I want.

Item: Person <<Changed To>> (null)

I understand I can do a custom comparer, but my question is: Can I do a custom comparer for a List? I.e.: If there was a list of Person (ICollection<Person>), can I create a custom comparer for that list only? And can I apply a custom comparer to only specific members?

**Comment by GregFinzer** [comment]
@Mansari 

Please see this page:
https://github.com/GregFinzer/Compare-Net-Objects/wiki/Comparing-Lists-of-Different-Lengths

**Comment by Mansari** [comment]
Thanks Greg. I ended up implementing a custom List comparer, based on the type of object included in the list I then do my comparison.

I built it based on your base ListComparer included in the source, with some modifications to cast the objects to IEnumerables and on finding the type of objects inside. I also made sure to only apply the custom list to specific types so it can continue to use the base comparers for others.

The library is very powerful once you know how it works and check the source. Thanks for creating such great work.
